File: libraries/adaptive-python/adaptive/src/adaptive/adaptive.py
from typing import Optional
from adaptive.resources.chat import Chat  # type: ignore
import os


class Adaptive:
    def __init__(self, api_key: Optional[str] = None, base_url: Optional[str] = None):
        """
        Initialize the Adaptive client.
        """

        # Read the API key from the environment variable if not provided

        if not api_key:
            api_key = os.getenv("ADAPTIVE_API_KEY")

        if not base_url:
            base_url = (
                "https://backend-go.salmonwave-ec8d1f2a.eastus.azurecontainerapps.io/"
            )
        self.chat = Chat(api_key=api_key, base_url=base_url)

File: adaptive-backend/internal/services/providers/adaptive/adaptive.go
package adaptive

import (
	"adaptive-backend/internal/services/providers/adaptive/chat"
	"adaptive-backend/internal/services/providers/provider_interfaces"
	"fmt"
	"os"

	"github.com/openai/openai-go"
	"github.com/openai/openai-go/option"
)

// AdaptiveService handles Adaptive API interactions using OpenAI-compatible SDK
type AdaptiveService struct {
	client *openai.Client
	chat   *chat.AdaptiveChat
}

// NewAdaptiveService creates a new Adaptive service using the OpenAI SDK
func NewAdaptiveService() (*AdaptiveService, error) {
	apiKey := os.Getenv("ADAPTIVE_API_KEY")
	if apiKey == "" {
		return nil, fmt.Errorf("ADAPTIVE_API_KEY environment variable not set")
	}

	baseURL := os.Getenv("ADAPTIVE_BASE_URL")
	if baseURL == "" {
		baseURL = "https://api.adaptive.ai/v1" // Default Adaptive API base URL
	}

	client := openai.NewClient(
		option.WithAPIKey(apiKey),
		option.WithBaseURL(baseURL),
	)

	chatService := chat.NewAdaptiveChat(&client)

	return &AdaptiveService{
		client: &client,
		chat:   chatService,
	}, nil
}

func (s *AdaptiveService) Chat() provider_interfaces.Chat {
	return s.chat
}

func (s *AdaptiveService) GetProviderName() string {
	return "adaptive"
}
